newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -33,20 +40,40 @@ const NumericProperty = (props: {|
   onUpdate: (newValue: string) => void,
 |}) => {
   const { properties, propertyName, step, onUpdate } = props;
-
+  const property = properties.get(propertyName);
   return (
     <SemiControlledTextField
       fullWidth
-      value={properties.get(propertyName).getValue()}
+      value={property.getValue()}
       key={propertyName}
-      floatingLabelText={properties.get(propertyName).getLabel()}
+      floatingLabelText={property.getLabel()}
       step={step}
       onChange={onUpdate}
       type="number"
+      endAdornment={<UnitAdornment property={property} />}
     />
   );
 };
 
+const UnitAdornment = (props: {| property: gdPropertyDescriptor |}) => {
+  const { property } = props;
+  const measurementUnit = property.getMeasurementUnit();
+  return (
+    <Tooltip
+      title={
+        <MarkdownText
+          source={getMeasurementUnitDescription(measurementUnit)}

```

Comment:
Maybe we should pass a react component instead of a markdown string to allow more precise layout?

Reply:
I did this too.